src/components/icon/icon.component.js

```diff
@@ -147,6 +149,8 @@ Icon.propTypes = {
   disabled: PropTypes.bool,
   /** Aria label for accessibility purposes */
   ariaLabel: PropTypes.string,
+  /** Allows the hover state */
+  hover: PropTypes.bool,
```

Comment:
I'd rather go with `disableHover` name for this blocking hover mechanism 
We would have hover normally as we have right now and after someone will put `disableHover` for the component then it will be blocked. It will also follow convention that we have when we are disabling some stuff like
`disableEscKey` in `alert/confirm/dialog`
Personally I think that `disableHover` is more understandable for the end user